doc/link_event_damping/Link-event-damping-HLD.md

```diff
@@ -0,0 +1,313 @@
+
+
+###  1.8. <a name='SAIAPI'></a>SAI API
+
+**SyncD Implementation**
+
+There are no changes expected in the SAI API.
+
+**Vendor SAI Implementation**
+
+To push the link event damping configuration, 2 new SAI port attributes needed:
+- SAI_PORT_ATTR_HOLD_TIME_UP
+- SAI_PORT_ATTR_HOLD_TIME_DOWN
+
+To get the total count of pre and post damped transitions on a port that reflect the actual count of link oper state change and count of link events that was advertized by SAI after link event damping, new counter attributes will be needed.
+
+###  1.9. <a name='Configurationandmanagement'></a>Configuration and management
+
+Link event damping related configurations are to be pushed via CONFIG DB for each interface.
+CONFIG DB has a port table for each interface and link event damping configuration will be added to the same table. Link event damping configuration will add 2 fields - ```hold_time_up``` and ```hold_time_down``` - in the CONFIG DB port table for each interface. The unit of these configuration parameters will be milliseconds.
+
+The lack of configuration on an interface - no ```hold_time_up``` and ```hold_time_down``` fields in the CONFIG DB port table - is interpreted as link event damping config disabled on the interface. This is treated the same as link event damping configuration with “hold_time_up = 0” and “hold_time_down = 0”.
+
+Only non-negative values for hold-time config should be supported.
+
+####  1.9.1. <a name='CLIEnhancements'></a>CLI Enhancements
+
+CLI tool should provide ability to configure the hold_time values.
+- Config hold_time up config to 5 seconds and hold time down config to 2 seconds:
+   ```sudo config interface hold_time_up hold_time_down Ethernet0 5000 2000```
+- Config hold_time up config to 10 seconds:
+  ```sudo config interface hold_time_up Ethernet0 10000```
+- Config hold_time up config to 0 seconds:
+  ```sudo config interface hold_time_down Ethernet0 0```
+
+####  1.9.2. <a name='YANGmodelEnhancements'></a>YANG model Enhancements
+
+The port yang model needs to update according to the config DB schema change. The yang model changes of new fields are described below:
+```
+leaf hold_time_up {
+  type uint32;
+}
+
+leaf hold_time_down {
+  type uint32;
+}
+```
+These new yang leaves will be added to sonic-port.yang.
+
+####  1.9.3. <a name='ConfigDBEnhancements'></a>Config DB Enhancements
+
+As mentioned above, link event damping configuration will add 2 fields - “hold_time_up” and “hold_time_down” - in the config DB PORT table for each interface.
+```
+; Defines information for port configuration
+key                     = PORT|port_name     ; configuration of the port
+; field                 = value
+...
+hold_time_up            = uint32             ; hold_time_up configuration
+hold_time_down          = uint32             ; hold_time_down configuration
+```
+
+- Config DB schema with link event damping config (hold-time up = 10 seconds and hold-time down = 5 seconds) in PORT table:
```

Comment:
please sync the word description and numbers in the example

Reply:
not valid anymore.